src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -5,6 +5,7 @@ import java.io.{ObjectInputStream, ObjectOutputStream}
 
 import scala.collection.mutable.{ArrayBuffer, ArrayBuilder, Builder, ArraySeq => MutableArraySeq}
 import scala.collection.{ArrayOps, ClassTagSeqFactory, SeqFactory, StrictOptimizedClassTagSeqFactory, View}
+import scala.collection._
```

Comment:
This appears to be what causes the build errors. Never import `collection._` in a more specific collection package. You want to use the names from `immutable`, in particular the build errors are due to `immutable.ArraySeq` now extending `collection.IndexedSeq` instead of `immutable.IndexedSeq`.